lib/ansible/module_utils/netconf.py

```diff
@@ -25,89 +25,58 @@
 # LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
-from contextlib import contextmanager
-
-from ansible.module_utils._text import to_bytes, to_text
-from ansible.module_utils.connection import exec_command
+from ansible.module_utils._text import to_text, to_native
+from ansible.module_utils.connection import Connection, ConnectionError
 
 try:
-    from lxml.etree import Element, SubElement, fromstring, tostring
+    from lxml.etree import Element, fromstring
 except ImportError:
-    from xml.etree.ElementTree import Element, SubElement, fromstring, tostring
+    from xml.etree.ElementTree import Element, fromstring
 
 NS_MAP = {'nc': "urn:ietf:params:xml:ns:netconf:base:1.0"}
 
 
-def send_request(module, obj, check_rc=True, ignore_warning=True):
```

Comment:
Why not keep `send_request()` as a convenient wrapper around `NetconfConnection.exec_rpc` like `exec_command()` in connection? There's a lot of churn related to replacing `send_request` calls to the same `conn = get_connection(); conn.execute_rpc(tostring(element), ignore_warning=False)` that could be handled here for every platform at once

Reply:
I thought about this, however, the problem is `get_connection()` is specific to the platform and is present in the corresponding module_utils file (eg. module_utils/junos.py) whereas send command will be a common code, so with above approach, we wouldn't know which platform `get_connection()` to call. Also, the connection object (module._connection) is instantiated after the first call to `get_connection()`

Reply:
You shouldn't need `get_connection()` for that, you can just make a new connection to the socket_path and let the gc pick it up it at the end of the scope like `exec_command()` does... there isn't any particular reason to care about which NetconfConnection instance gets the message as far as I can tell.

Reply:
`NetconfConnection` has additional logic to parse XML response to check if error severity in response is warning or error. For `Connection` the reply directly passed to module code without extra processing.

Reply:
I _think_ we're on the same page now. I initially thought about it as a simple wrapper just as `exec_command` is, but in the end it would have the same signature as `send_request`, so keeping the name would remove a lot of changes from the modules.

Then again, we don't have any external consumers of the netconf plugin to my knowledge, so we're not tied to names as much as with network_cli, and renaming the function to make ithe intent clearer might be beneficial.

Reply:
Fixed